lamindb/_query_set.py

```diff
@@ -69,8 +71,33 @@ def one_helper(self):
         return self[0]
 
 
-def process_expressions(registry: Registry, expressions: dict) -> dict:
-    if registry in {Artifact, Collection}:
+def process_expressions(queryset: QuerySet, expressions: dict) -> dict:
+    def _map_databases(value: Any, key: str, target_db: str) -> tuple[str, Any]:
+        if isinstance(value, Record):
+            if value._state.db != target_db:
```

Comment:
I thought of this but I don't think this can occur. You can't query with a non-saved record because the query is mediated with the pk and an unsaved record doesn't have a pk.

The tests showed that `target_db` is sometimes `None` and the whole thing isn't even called.